.github/workflows/e2e_libvirt.yaml

```diff
@@ -35,8 +45,102 @@ jobs:
           name: ${{ inputs.install_directory_artifact }}
           path: install
 
+      - name: Install Libvirt
+        run: |
+          set -o pipefail
+          echo "::group::Install packages"
+          sudo apt-get update
+          sudo apt-get install -y qemu-kvm libvirt-daemon-system libvirt-dev
+          echo "::endgroup::"
+          kvm-ok
+          # Create the default storage pool if not defined.
+          echo "::group::Setup Libvirt default storage pool"
+          if ! sudo virsh pool-list --all | grep default >/dev/null; then
+            sudo virsh pool-define-as default dir - - - - "/var/lib/libvirt/images"
+            sudo virsh pool-build default
+          fi
+          sudo virsh pool-start default || true
+          echo "::endgroup::"
+
+          sudo setfacl -m "u:${USER}:rwx" /var/lib/libvirt/images
+          sudo adduser "$USER" libvirt
+          # Although it adds the runner user to libvirt's group, it is getting
+          # hard to re-load the actual session so that it takes effect. As
+          # an alternative, let's set the permission on the libvirt's socket
+          # file directly.
+          sudo setfacl -m "u:${USER}:rwx" /var/run/libvirt/libvirt-sock
+
+      - name: Install kcli
+        run: |
+          echo "::group::Install dependencies"
+          sudo apt-get install -y genisoimage
+          echo "::endgroup::"
+
+          echo "::group::Install kcli"
+          curl https://raw.githubusercontent.com/karmab/kcli/main/install.sh | sudo bash
+          echo "::endgroup::"
+
+          # kcli needs a pair of keys to setup the VMs
+          [ -f ~/.ssh/id_rsa ] || \
+            ssh-keygen -t rsa -f ~/.ssh/id_rsa -N ""
+
+          # Newest version of kcli does not index old Ubuntu images like 20.04
+          echo "::group::Download Ubuntu 20.04 image"
+          kcli download image -u https://cloud-images.ubuntu.com/releases/20.04/release/ubuntu-20.04-server-cloudimg-amd64.img ubuntu2004
+          echo "::endgroup::"
+
+        # This is the key used by cloud-api-adaptor to connect to libvirt
+      - name: Generate the SSH key
+        run: |
+          ssh-keygen -f ./id_rsa -N ""
```

Comment:
@wainersm Shouldn't this be a required step for `Install kcli` rather than handling ssh-keygen there as well?

Reply:
hi @ldoktor , I'm glad to see you here :)

This key is used by cloud-api-adaptor to connect to the Libvirt on host. It could be the same key as used by the kcli to install the cluster, but I prefer to keep them two separated keys.